### PR TITLE
[xmlsec]Update to 1.2.32

### PR DESCRIPTION
**Pull request by SvenPStarFinanz**
- #### What does your PR fix?  
Updates xmlsec to 1.2.32. Among other minor fixes that version removes the need to patch UWP build.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
no change

- #### Does your PR follow the [maintainer guide]
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

### CODE REVIEW

File: ports/xmlsec/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "xmlsec",
-  "version-string": "1.2.31",
-  "port-version": 1,
+  "version-string": "1.2.32",
```
Comment (JonLiu1993) [inline ports/xmlsec/vcpkg.json:3]: @SvenPStarFinanz ,We should use "version" replace "version-string"

File: versions/x-/xmlsec.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "369cbf67956067b49be4de045129b9cfbfed9431",
+      "version-string": "1.2.32",
```
Comment (JonLiu1993) [inline versions/x-/xmlsec.json:5]: We should use "version" replace "version-string"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/xmlsec/vcpkg.json (hunk 1 above)
Comment (SvenPStarFinanz) [reply ports/xmlsec/vcpkg.json]: Ok, did so.

File: versions/x-/xmlsec.json (hunk 2 above)
Comment (SvenPStarFinanz) [reply versions/x-/xmlsec.json]: Did so.

**Comment by SvenPStarFinanz** [comment]
Apart from the version update I found/fixed a problem resulted by #16908 on Windows build of xmlsec. In src\dl.c of xmlsec there was constructed the name of the crypto library, which always have to begin with "lib", so I turned back to the old file name on Windows.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
That makes sense to me, thanks for the update & fix!

**Pull request merged**